### PR TITLE
update open api client and types

### PR DESCRIPTION
**Pull request by xstelea**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -18,7 +18,6 @@ module.exports = {
 		'plugin:functional/external-recommended', // https://github.com/jonaskello/eslint-plugin-functional
 		'plugin:functional/recommended', // https://github.com/jonaskello/eslint-plugin-functional
 		'prettier/@typescript-eslint', // Uses eslint-config-prettier to disable ESLint rules from @typescript-eslint/eslint-plugin that would conflict with prettier
-		'plugin:prettier/recommended', // Enables eslint-plugin-prettier and eslint-config-prettier. This will display prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
```
Comment (AbstractFruitFactory) [inline .eslintrc.js:21]: reasoning?

File: packages/application/src/api/open-api/interface.ts [hunk 2]
```diff
@@ -22,29 +22,35 @@ import {
...
+			() => call(method, params),
+			result => result.mapErr(e => [e]).andThen(handleResponse),
+		)()
 
 export const getAPI = pipe(
 	(call: OpenApiClientCall) => callAPIWith(call),
 
 	callAPI => ({
 		gateway: callAPI('gatewayPost')(handleGatewayResponse),
+		// @ts-ignore: TODO
```
Comment (AbstractFruitFactory) [inline packages/application/src/api/open-api/interface.ts:44]: whats going on here?

File: .eslintrc.js (hunk 1 above)
Comment (xstelea) [reply .eslintrc.js:21]: had some conflicts between my local prettier config and eslint prettier. Disabled it during development and accidentally pushed it

File: packages/application/src/api/open-api/interface.ts (hunk 2 above)
Comment (xstelea) [reply packages/application/src/api/open-api/interface.ts]: had some misbehaving types. a WIP thing

**Pull request merged**